### PR TITLE
Sklearn adapter 683

### LINKED ISSUE #683: [OpenSTEF 4.0] Create adapter for sklearn transforms
As OpenSTEF user
I want to be able to use the sklearn transforms
So that I can quickly add new transforms to the pipelines

> [!IMPORTANT]
> Changes should be done by forking [OpenSTEF 4.0.0 tracking branch](https://github.com/OpenSTEF/openstef/tree/release/v4.0.0) (`release/v4.0.0`) and then make a PR back onto `release/v4.0.0` branch.
> Please read the [contributing guide](https://openstef.github.io/openstef/v4/contribute/index.html) before you start.

Definition of Done:

✅ Acceptance criteria

- [ ] Adapter has been written to implement [TimeSeriesTransform](https://github.com/OpenSTEF/openstef/blob/release/v4.0.0/packages/openstef-core/src/openstef_core/transforms/dataset_transforms.py#L31) so it can be used in FeatureEngineeringPipeline
- [ ] Any sklearn transform can be adapted to OpenSTEF 4.0 

📄 Documentation criteria:

Interfaces with other products and services

Update relevant confluence pages

🧪 Test criteria:

Unit tests written and passed

⌛ Dependencies:

N/A

🚀 Releasing:

N/A

Other information:

🌍 Background

### PR DESCRIPTION
**Pull request by ChhayashreeDesai**
## Description
This PR adds a `SklearnTransformerAdapter` class that allows OpenSTEF to use any scikit-learn transformer in its pipelines.

## Changes
- Added `SklearnTransformerAdapter` in `openstef/feature_engineering/`
- Updated `__init__.py` to export the new class
- Added comprehensive unit tests with 100% coverage

## Fixes #683

-My code follows the code style of this project
-I have added tests that prove my fix is effective
-New and existing unit tests pass locally with my changes

## Testing
- **All 84 existing tests pass** - no regression
- New adapter tests pass with 100% coverage
- Adapter works with various sklearn transformers (StandardScaler, MinMaxScaler, etc.)
- Preserves DataFrame structure including columns and index
- Code follows OpenSTEF style guidelines (flake8 compliant)

## Example Usage
```python
from sklearn.preprocessing import StandardScaler, MinMaxScaler
from openstef.feature_engineering import SklearnTransformerAdapter

# Use any sklearn transformer in OpenSTEF pipelines
scaler_adapter = SklearnTransformerAdapter(StandardScaler)
normalizer_adapter = SklearnTransformerAdapter(MinMaxScaler, feature_range=(0, 1))

### CODE REVIEW

**Comment by egordm** [comment]
Hi, thanks for your contribution!

I see you have picked up issue #683 in this PR. That issue is tagged as part of OpenSTEF 4.0 so the PR should be made onto [release/v4.0.0](https://github.com/OpenSTEF/openstef/tree/release/v4.0.0) branch.

Also looking at the code currently the code implements SKLearn transform interfaces and forwards it to SKLearn objects.
In the in v4.0.0 branch we have our own transform interface which should be implemented and from which the calls should be forwarded to sklearn transforms.

See [TimeSeriesTransform](https://github.com/OpenSTEF/openstef/blob/release/v4.0.0/packages/openstef-core/src/openstef_core/transforms/dataset_transforms.py#L31).

I have added more info to the issue accordingly as well.

If possible could you update your branch with these changes? Thanks.
